src/components/FeatureViews/Scheduler/Scheduler.js

```diff
@@ -200,38 +243,83 @@ class Scheduler extends Component {
     changeView = (value) => {
         let calendarApi = this.calendarComponentRef.current.getApi();
         calendarApi.changeView(value);
+        let filter;
+
         let date = this.currentDate();
-        this.setState({
-            "viewValue": value,
-            "currentDate": date,
+        this.setState(() => {
+            this.props.calendarActions.fetchSessions({
+                config: {
+                    params: {
+                        time_frame: filter,
+                        view_option: this.viewOptions[this.state.calendarFilterValue],
+                    }
+                }
+            });
+            return {
+                "viewValue": value,
+                "currentDate": date,
+            }
         });
     }
 
     goToNext = () => {
         let calendarApi = this.calendarComponentRef.current.getApi();
-
         calendarApi.next();
         let date = this.currentDate();
-        this.setState({
-            "currentDate": date,
+        this.props.calendarActions.fetchSessions({
+            config: {
+                params: {
+                    time_frame: this.calendarViewToFilterVal[this.state.viewValue],
+                    view_option: this.viewOptions[this.state.calendarFilterValue],
+                    time_shift: this.state.timeShift + 1,
+                }
+            }
+        });
+        this.setState((state) => {
+            return {
+                "currentDate": date,
+                "timeShift": state.timeShift + 1,
+            }
         });
```

Comment:
Can be shortened to:

    this.setState((state) => ({
        // object properties here
    }));